DM-49735: Use uv for uploads

rra opened the pull request:
The GitHub Action to which we previously delegated the PyPI uploads cannot be called from a composite action when using trusted environments. Since uv now has built-in support for building and uploading packages, switch to uv for both steps, dropping use of both the PyPA build tool and twine. This significantly simplifies the action and removes the need for the `python-version` parameter.

In the interests of removing twine from the workflow entirely, this drops the twine check step, which was previously called when uploading was disabled. Per the twine documentation, that step only checked whether the package long description can be rendered.

rra committed:
Use uv for uploads

The GitHub Action to which we previously delegated the PyPI uploads
cannot be called from a composite action when using trusted
environments. Since uv now has built-in support for building and
uploading packages, switch to uv for both steps, dropping use of both
the PyPA build tool and twine. This significantly simplifies the
action and removes the need for the `python-version` parameter.

In the interests of removing twine from the workflow entirely, this
drops the twine check step, which was previously called when uploading
was disabled. Per the twine documentation, that step only checked
whether the package long description can be rendered.

jonathansick commented on README.md:
```diff
@@ -4,14 +4,16 @@ This is a [composite GitHub Action](https://docs.github.com/en/actions/creating-
+
+Both actions are done with [uv](https://docs.astral.sh/uv/).
 
 > **Note**
```
```suggestion
> [!NOTE]
```

We can use the new-ish [GitHub Markdown alerts syntax now](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).

jonathansick reviewed (approved):
Awesome 🚀 I agree the twine check doesn't seem particularly useful at this point.

rra committed:
Use the new GitHub alerts Markdown extension

In `README.md`, mark notes and warnings with the GitHub alerts
Markdown extension.

fajpunk commented on README.md:
```diff
@@ -4,14 +4,16 @@ This is a [composite GitHub Action](https://docs.github.com/en/actions/creating-
-> Since this action uses the [build](https://pypa-build.readthedocs.io/en/stable/index.html) tool, any [PEP 517](https://peps.python.org/pep-0517/)-compatible packaging backend is supported, including [setuptools](https://setuptools.pypa.io/en/latest/).
-> You can make your project PEP 517-compatible by adding a `project.toml` file and specifying the build backend in a `[build-system]` section ([see setuptools' tutorial on this](https://setuptools.pypa.io/en/latest/build_meta.html)).
+> Packages are built using the `--no-sources` option as recommended by the [uv documentation](https://docs.astral.sh/uv/guides/package/#building-your-package) to ensure that uv-specific configuration is not required.
+> Packages must be compatible with PEP 517, which means they must have a `project.toml` file that specifies the build backend in a `[build-system]` section ([see setuptools' tutorial on this](https://setuptools.pypa.io/en/latest/build_meta.html)).
```
```suggestion
> Packages must be compatible with PEP 517, which means they must have a `pyproject.toml` file that specifies the build backend in a `[build-system]` section ([see setuptools' tutorial on this](https://setuptools.pypa.io/en/latest/build_meta.html)).
```
